### PR TITLE
feat: Support Braze Web SDK V5

### PR DESCRIPTION
**Pull request by rmi22186**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
Upgrading Braze web kit to v5. Changelog is here https://www.braze.com/docs/developer_guide/platform_integration_guides/web/changelog.  The only item from the changelog that is relevant to us is changing the option `enableHtmlInAppMessages` to be `allowUserSuppliedJavascript`.

Otherwise, no breaking changes.  Braze did update their push code, so I have update our code to reflect it and updated the link in the comments to Braze's docs.

 ## Testing Plan
 - [X] Was this tested locally? If not, explain why.
Tested commerce and logged events.  Successfully tested push notifications on chrome and firefox to ensure the push code is still working.

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-6069

### CODE REVIEW

File: src/BrazeKit-dev.js [hunk 1]
```diff
@@ -666,27 +666,31 @@ var constructor = function () {
...
+                // check the value of our key `msg-id` defined in the Braze dashboard
+                if (keyValuePairs['msg-id'] === 'push-primer') {
                     pushPrimer = true;
-                    // We don't want to display the soft push prompt to users on browsers that don't support push, or if the user
-                    // has already granted/blocked permission
+                    // We don't want to display the soft push prompt to users on browsers
+                    // that don't support push, or if the user has already granted/blocked permission
                     if (
-                        !braze.isPushSupported() ||
+                        braze.isPushSupported() === false ||
```
Comment (alexs-mparticle) [inline src/BrazeKit-dev.js:682]: I think it might be cleaner to define a helper method called `isPushSupported()` that checks all of these values.

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]

File: src/BrazeKit-dev.js (hunk 1 above)
Comment (rmi22186) [reply src/BrazeKit-dev.js:682]: While I agree, this is copied and pasted from their docs, so I want to make as minimal changes as possible unless it deals with us adding certain flags, like our variables for `shouldDisplay` and `pushPrimer`

**Review by mmustafa-tse — APPROVED** [review]
LGTM! nice catch on removing the deprecated `enableHtmlInAppMessages` and replacing it with `allowUserSuppliedJavascript` per their change log

File: src/BrazeKit-dev.js (hunk 1 above)
Comment (rmi22186) [reply src/BrazeKit-dev.js:682]: given this is the only comment, and @mmustafa-tse has approved, i am going to merge

**Pull request merged**